Assets/Dev/B/Script/CardSystem.cs

```diff
@@ -80,10 +82,14 @@ public void PlayCard(int index)
 
     public void DrawCard()
     {
-        if (lastIndex != maxCardCount)
+        if (turnSystem.GetBattleStatus() == BattleStatus.PlayerCombat)
         {
-            InstantiateCard(lastIndex);
-            lastIndex++;
+            turnSystem.NextTurn();
```

Comment:
This looks problematic. Drawing a card shouldn't govern what happens in the turn system, and it looks like it's changing the turn before doing anything. When I draw a card, I'm getting a message that says 'select other tiles.' I think this is because the turn is going to enemy move

Reply:
thats right, but I dont see the prolem here